rtnetlink/src/link/add.rs

```diff
@@ -17,6 +17,137 @@ use crate::{
     Handle,
 };
 
+/// A request to create a new vxlan link. This is equivalent to the `ip link add NAME vxlan id ID ...` commands.
+///
+/// It provides methods to customize the creation of
+/// the vxlan interface
+pub struct VxlanAddRequest {
+    request: LinkAddRequest,
+    info_data: Vec<InfoVxlan>,
+}
+
+impl VxlanAddRequest {
+    pub async fn execute(self) -> Result<(), Error> {
+        let s = self
+            .request
+            .link_info(InfoKind::Vxlan, Some(InfoData::Vxlan(self.info_data)));
+        s.execute().await
+    }
+
+    pub fn up(mut self) -> Self {
+        self.request = self.request.up();
+        self
+    }
+
+    pub fn link(mut self, index: u32) -> Self {
+        self.info_data.push(InfoVxlan::Link(index));
+        self
+    }
+
+    pub fn port(mut self, port: u16) -> Self {
+        self.info_data.push(InfoVxlan::Port(port));
+        self
+    }
+
+    pub fn group(mut self, addr: std::net::Ipv4Addr) -> Self {
```

Comment:
Would you mind addind some doc to these methods? It doesn't need to be an in-depth explanation of VXLAN, just a couple sentences would be nice, or the equivalent `ip` command.